### PR TITLE
init OSAVersion

### PR DESCRIPTION
**Pull request by burnout87**
Initialized as a Name parameter

### CODE REVIEW

**Comment by burnout87** [comment]
This only tackles the initialization of a OSAVersion parameter. An additional change would ideally be required at dispatcher level.

**Review by volodymyrss — CHANGES_REQUESTED** [review]
cannt check closely. Well it fails CI test. Coverage of this might needs to be improved.

File: cdci_osa_plugin/osa_common_pars.py [hunk 1]
```diff
@@ -106,18 +107,20 @@ def __init__(self,
...
+        allowed_values = []
         if os.environ.get('DISPATCHER_MOCK_KB', 'no') != 'yes':
             # this is in addition to base OSA versions
-            self._allowed_values = get_osa_versions()
+            allowed_values = get_osa_versions()
         else:
-            self._allowed_values = ["OSA11.0-dev210827.0528-37487"]
+            allowed_values = ["OSA11.0-dev210827.0528-37487"]
+
+        super(OSAVersion, self).__init__(value=value, name=name, allowed_values=allowed_values)
```
[the remark this answers is not in the document]
Comment (burnout87) [reply cdci_osa_plugin/osa_common_pars.py:122]: Thanks for the tip!

**Comment by burnout87** [comment]
> This only tackles the initialization of a OSAVersion parameter. An additional change would ideally be required at dispatcher level.

Just noticed that also for the time parameters (eg T and T2) units is not set, and it looks like it should be the related format, also this, should be tacked at a dispatcher-app level.

File: cdci_osa_plugin/osa_common_pars.py [hunk 2]
```diff
@@ -106,18 +103,14 @@ def __init__(self,
         else:
             self._allowed_base_osa_version_values = allowed_base_osa_version_values
 
-        self.name = name
-        self.value = value
-
-        self.units_name = "string"
+        super().__init__(value=value, name=name,)
```
Comment (volodymyrss) [inline cdci_osa_plugin/osa_common_pars.py:106]: extra coma at the end

**Comment by volodymyrss** [comment]
the tests for ISGRI and JEMX are almost identical. Could they be combined with parametrization?

**Comment by burnout87** [comment]
> the tests for ISGRI and JEMX are almost identical. Could they be combined with parametrization?

I thought the same, but I went for this approach since we already had two instrument specific test modules.

**Comment by volodymyrss** [comment]
> > the tests for ISGRI and JEMX are almost identical. Could they be combined with parametrization?
> 
> I thought the same, but I went for this approach since we already had two instrument specific test modules.

We also have test groups not specific to instruments.

Why further increase duplication, even if there is some already. For whatever reason it was done before. It's easier to maintain code without duplication, in case of changes it's not necessary to duplicate them too.

I do not suppose it's any more difficult to do it in two tests instead of four?
Well, if it's not convenient now, we can leave a note and do it later.

**Comment by burnout87** [comment]
> > > the tests for ISGRI and JEMX are almost identical. Could they be combined with parametrization?
> > 
> > 
> > I thought the same, but I went for this approach since we already had two instrument specific test modules.
> 
> We also have test groups not specific to instruments.
> 
> Why further increase duplication, even if there is some already. For whatever reason it was done before. It's easier to maintain code without duplication, in case of changes it's not necessary to duplicate them too.
> 
> I do not suppose it's any more difficult to do it in two tests instead of four? Well, if it's not convenient now, we can leave a note and do it later.

and now they are

**Review by volodymyrss — APPROVED** [review]
cool, thanks!

**Pull request merged**